### PR TITLE
Bug 1777486: Update the monitoring and logging URLs name

### PR DESCRIPTION
**Pull request by jhadvig**
/assign @benjaminapetersen

### CODE REVIEW

**Comment by jhadvig** [comment]
we need backport this PR  to 4.4 as well as https://github.com/openshift/console-operator/pull/381 ... right ?

**Comment by spadgett** [comment]
> we need backport this PR to 4.4 as well as #381 ... right ?

Are the monitoring and logging teams backporting the change to add the config maps to openshift-config-managed?

File: pkg/console/subresource/configmap/configmap_test.go [hunk 1]
```diff
@@ -410,16 +410,16 @@ providers:
...
 				monitoringSharedConfig: &corev1.ConfigMap{
 					Data: map[string]string{
-						"alertmanagerURL": "https://alertmanager.url.com",
-						"grafanaURL":      "https://grafana.url.com",
-						"prometheusURL":   "https://prometheus.url.com",
-						"thanosURL":       "https://thanos.url.com",
+						"alertmanagerPublicURL": "https://alertmanager.url.com",
+						"grafanaPublicURL":      "https://grafana.url.com",
+						"prometheusPublicURL":   "https://prometheus.url.com",
+						"thanosPublicURL":       "https://thanos.url.com",
```
Comment (spadgett) [inline pkg/console/subresource/configmap/configmap_test.go]: Are the other teams making this change to the config maps they own?

Comment (jhadvig) [reply pkg/console/subresource/configmap/configmap_test.go]: yes, both team shall do so

**Comment by jhadvig** [comment]
@spadgett per discussion with them, they should backport.

**Comment by jhadvig** [comment]
FYI the naming fix thats adding the "Public" string into the key names has merged for the logging operator

**Comment by jhadvig** [comment]
@benjaminapetersen PR updated

**Comment by simonpasquier** [comment]
this is the related monitoring PR: https://github.com/openshift/cluster-monitoring-operator/pull/677

**Comment by jhadvig** [comment]
@benjaminapetersen I think this one is to go :)

**Comment by benjaminapetersen** [comment]
Actually, you have a unit test fail:

```
 --- FAIL: TestDefaultConfigMap (0.01s)
    --- FAIL: TestDefaultConfigMap/Test_default_configmap,_no_customization (0.00s)
        configmap_test.go:510: [map[loggingInfo]: <does not have key> != map[] map[monitoringInfo]: <does not have key> != map[]]
    --- FAIL: TestDefaultConfigMap/Test_configmap_with_default-ingress-cert (0.00s)
        configmap_test.go:510: [map[loggingInfo]: <does not have key> != map[] map[monitoringInfo]: <does not have key> != map[]]
    --- FAIL: TestDefaultConfigMap/Test_managed_config_to_override_default_config (0.00s)
        configmap_test.go:510: [map[loggingInfo]: <does not have key> != map[] map[monitoringInfo]: <does not have key> != map[]]
    --- FAIL: TestDefaultConfigMap/Test_operator_config_overriding_default_config_and_managed_config (0.00s)
        configmap_test.go:510: [map[loggingInfo]: <does not have key> != map[] map[monitoringInfo]: <does not have key> != map[]]
    --- FAIL: TestDefaultConfigMap/Test_operator_config_with_Custom_Branding_Values (0.00s)
        configmap_test.go:510: [map[loggingInfo]: <does not have key> != map[] map[monitoringInfo]: <does not have key> != map[]]
    --- FAIL: TestDefaultConfigMap/Test_operator_config_with_Statuspage_pageID (0.00s)
        configmap_test.go:510: [map[monitoringInfo]: <does not have key> != map[] map[loggingInfo]: <does not have key> != map[]]
FAIL
FAIL	github.com/openshift/console-operator/pkg/console/subresource/configmap	0.019s 
```

**Comment by jhadvig** [comment]
PR update.. since we remove the `omitempty` I had to add the empty map in the test structs for `monitoringInfo` and `loggingInfo`

**Comment by spadgett** [comment]
> PR update.. since we remove the `omitempty` I had to add the empty map in the test structs for `monitoringInfo` and `loggingInfo`

Is there a reason to remove `omitempty` here?

**Comment by spadgett** [comment]
/hold
It looks like there's an issue moving the logging config to openshift-config-managed. We might have to read it from the logging namespace.

**Comment by jhadvig** [comment]
@spadgett the issue with the `omitempty` is the parity with console server types. https://github.com/openshift/console/pull/4309#discussion_r383887293

**Comment by spadgett** [comment]
> @spadgett the issue with the `omitempty` is the parity with console server types. [openshift/console#4309 (comment)](https://github.com/openshift/console/pull/4309#discussion_r383887293)

Sorry I didn't realize that @benjaminapetersen had gone through and added `omitempty` to all the types in console. I thought we weren't using it there. I think we want it in that case.

**Comment by jhadvig** [comment]
@spadgett I would agree to remove the `omitempty` since thats what we are doing with other fields.
@benjaminapetersen any objections ?

**Comment by spadgett** [comment]
> @spadgett I would agree to remove the `omitempty` since thats what we are doing with other fields.

I guess I'm not clear. We have `omitempty` on the other fields. I was proposing adding it back to the new properties in your PR.

Either way, we should be internally consistent in its usage in the console repo. I don't think it matters much either way since we only read the config, never serialize.

File: pkg/console/subresource/consoleserver/types.go [hunk 2]
```diff
@@ -48,16 +48,16 @@ type ClusterInfo struct {
 
 // Monitoring holds URLs for monitoring related services
 type MonitoringInfo struct {
-	AlertmanagerURL string `yaml:"alertmanagerURL,omitempty"`
-	GrafanaURL      string `yaml:"grafanaURL,omitempty"`
-	PrometheusURL   string `yaml:"prometheusURL,omitempty"`
-	ThanosURL       string `yaml:"thanosURL,omitempty"`
+	AlertmanagerPublicURL string `yaml:"alertmanagerPublicURL,omitempty"`
```
Comment (benjaminapetersen) [inline pkg/console/subresource/consoleserver/types.go]: I would use `omitempty` here, as the others.

File: pkg/console/subresource/consoleserver/types.go [hunk 3]
```diff
@@ -18,8 +18,8 @@ type Config struct {
 	Auth           `yaml:"auth"`
 	Customization  `yaml:"customization"`
 	Providers      `yaml:"providers"`
-	MonitoringInfo `yaml:"monitoringInfo,omitempty"`
-	LoggingInfo    `yaml:"loggingInfo,omitempty"`
+	MonitoringInfo `yaml:"monitoringInfo"`
```
Comment (benjaminapetersen) [inline pkg/console/subresource/consoleserver/types.go]: I would leave of `omitempty` here, as the others props at this level don't have it.

Comment (benjaminapetersen) [reply pkg/console/subresource/consoleserver/types.go]: These all should have it, but it was never fixed.

Comment (spadgett) [reply pkg/console/subresource/consoleserver/types.go]: > These all should have it, but it was never fixed.

We'd need to change them to pointers for `omitempty` to work iirc

**Comment by jhadvig** [comment]
@benjaminapetersen @spadgett comments addressed. PTAL

**Comment by jhadvig** [comment]
@benjaminapetersen PR update per our discussion.PTAL

**Comment by benjaminapetersen** [comment]
/cherry-pick cancel

(maybe?)

**Pull request merged**